haralyzer/mixins.py

```diff
@@ -0,0 +1,42 @@
+"""Mixin Objects that allow for shared methods"""
+
+
+class GetHeaders(object):
+    """Mixin to get a header"""
+    def get_header_value(self, name):
+        """
+        Returns the header value of the header defined in ``name``
+
+        :param name: ``str`` name of the header to get the value of
+        """
+        for x in self.raw_entry["headers"]:
+            if x["name"].lower() == name.lower():
+                return x["value"]
+
+
+class MimicDict(object):

```

Comment:
I like where you are headed with the `MimicDict` base class, but instead of manually implementing methods like `.get` perhaps we should be using an abstract base class like `MutableMapping` which would give you these methods "for free" and also ensure that all required dunder methods are implemented. Something along the lines of [this](https://stackoverflow.com/a/3387975/1366989) maybe?

Reply:
It actually looks really easy with the MutableMapping.
Using:
```
class MimicDict(MutableMapping, ABC):
    """Mixin for functions to mimic a dictionary for backward compatibility"""

    def __getitem__(self, item):
        return self.raw_entry[item]

    def __len__(self):
        return len(self.raw_entry)

    def __delitem__(self, key):
        del self.raw_entry[key]

    def __iter__(self):
        return iter(self.raw_entry)

    def __setitem__(self, key, value):
        self.raw_entry[key] = value
```
Passes all current tests